pkg/k8sutil/k8sutil.go

```diff
@@ -409,6 +409,11 @@ func buildStatefulSet(statefulSetName, clusterName, deploymentType, baseImage, s
 		scheme = v1.URISchemeHTTP
 	}
 
+	// add a nodeselector for failure-domain.beta.kubernetes.io/zone:  based on the AZ. Use storageClass var as that contains the current AZ
+	if storageClass != "default" {
+		nodeSelector["failure-domain.beta.kubernetes.io/zone"] = storageClass
```

Comment:
Someone could specify a storageClass that's not `default` and this would break.

Reply:
This may not be new to your PR, but could be a bug for someone. Let's open a new issue if not to fix.

Reply:
This is no longer needed for me and I have removed it entirely